monai/transforms/utils.py

```diff
@@ -582,7 +582,7 @@ def weighted_patch_samples(
     if not v[-1] or not isfinite(v[-1]) or v[-1] < 0:  # uniform sampling
         idx = r_state.randint(0, len(v), size=n_samples)
     else:
-        r, *_ = convert_to_dst_type(r_state.random(n_samples), v)
+        r, *_ = convert_to_dst_type(r_state.random(n_samples), v, structure_only=True)
```

Comment:
```suggestion
        r_samples = r_state.random(n_samples)
        r, *_ = convert_to_dst_type(r_samples, v, dtype=r_samples.dtype)
```

Would this not be equivalent to what you're wanting to accomplish? I think `convert_to_dst_type` is getting rather complicated now with various options. In the docstring you provide with `structure_only` it's not clear what the implications are of using it, you could say more exlicitly that the dtype will be that of `src` if it has one or `dtype` if not (then the user has to realise the `dtype` will be taken from `dest` if this is `None`). If my suggestion works for you we should avoid adding this complexity.

Reply:
You’re right, this does have the same effect. I shouldn’t have made the function more complex. The current logic already meets the requirements, and I’ve made the necessary adjustments.